scripts/check_blocking_repos.py

```diff
@@ -46,12 +54,24 @@ def is_released(repo, dist_file):
 try:
     i = valid_distro_keys.index(distro_key)
 except ValueError:
-    print('Distribution key not found in list of valid distributions.')
+    print('Distribution key not found in list of valid distributions.', file=sys.stderr)
     exit(-1)
-if i == 0:
-    print('No previous distribution found.')
+if i == 0 and not args.comparison:
+    print('No previous distribution found.', file=sys.stderr)
+    exit(-1)
+
+if args.comparison:
+    if args.comparison not in valid_distro_keys:
+        print('Invalid rosdistro [%s] selected for comparison.' % args.comparison, file=sys.stderr)
+        print('Valid rosdistros are %s.' % valid_distro_keys, file=sys.stderr)
+        exit(-1)
+    prev_distro_key = args.comparison
+else:
+    prev_distro_key = valid_distro_keys[i - 1]
+
+if prev_distro_key == distro_key:
+    print('It does not make sense to compare distro [%s] to itself.', file=sys.stderr)
```

Comment:
Having this check in a wider scope while it is only relevant in a smaller scope is imo not following out programming conventions to "use the narrowest scope possible".